lv_objx/lv_cpicker.c

```diff
@@ -417,129 +417,147 @@ static bool lv_cpicker_disc_design(lv_obj_t * cpicker, const lv_area_t * mask, l
         lv_coord_t y = cpicker->coords.y1 + lv_obj_get_height(cpicker) / 2;
         lv_opa_t opa_scale = lv_obj_get_opa_scale(cpicker);
 
+        uint8_t redraw_wheel = 0;
+
+        lv_area_t center_ind_area;
+
+        uint32_t rin = r - styleCopy.line.width;
+        //the square area (a and b being sides) should fit into the center of diameter d
+        //we have:
+        //a^2+b^2<=d^2
+        //2a^2 <= d^2
+        //a^2<=(d^2)/2
+        //a <= sqrt((d^2)/2)
+        uint16_t radius = sqrt((4*rin*rin)/2)/2 - style->body.padding.inner;
```

Comment:
@AloyseTech This forces the user of the color picker to have a math library available, which may not always be the case. It also makes the color picker depend on floating point rather than integer math.

Reply:
I will use a custom fast_sqrt at the end. This is for experimentation purpose only.